### PR TITLE
Refactoring reports creation for clarity

### PR DESCRIPTION
**Pull request by mwatts15**
This is a refactor to hopefully improve clarity in what `Report.save` does. I've replaced comments with method names where it seemed appropriate and introduced objects to track some variables that were tightly coupled together already. There was also some needless repetition that was reduced.

### CODE REVIEW

File: codespeed/models.py [hunk 1]
```diff
@@ -73,6 +76,31 @@ def save(self, *args, **kwargs):
         super(Project, self).save(*args, **kwargs)
 
 
+class HistoricalValue(object):
+    def __init__(self, name=None, val=0, color='none'):
+        self.name = name
+        self.val = val
+        self.color = color
+
+    def if_significant_update_to(self, val, color, name):
```
Comment (tobami) [inline codespeed/models.py]: I find the name not easy to read. How about "update_if_significant_change" or "update_to_biggest_change"?

Comment (mwatts15) [inline codespeed/models.py]: I agree, but what about 'update_if_less_important_than'? It reads pretty well at the point of usage:

    max_trend.update_if_less_important_than(val, color, benchmark_name)

**Comment by tobami** [comment]
"less_" or "more_"?

**Comment by mwatts15** [comment]
@tobami "less", because we're asking `max_trend` to update itself if it's less important than `(val, color, benchmark_name)`

**Comment by mwatts15** [comment]
@tobami also, neglected to mention, I am working on a test.

**Comment by tobami** [comment]
I was hesitating because it is still a bit too complex for my taste, but hey, you improved it by like, a lot, and added tests to it, so let's merge this big win.

**Pull request merged**